### PR TITLE
Image added

### PR DESCRIPTION
**Pull request by ShadyNagy**
- Upload image added in Create and Edit.
- Remove button added in Create and Edit.

### CODE REVIEW

File: src/PublicApi/ImageValidators.cs [hunk 1]
```diff
@@ -0,0 +1,25 @@
+﻿using System;
+using System.IO;
+
+namespace Microsoft.eShopWeb.PublicApi
+{
+    public static class ImageValidators
+    {
+        private const int ImageMinimumBytes = 512000;
```
Comment (ardalis) [inline src/PublicApi/ImageValidators.cs:8]: Maximum

File: src/PublicApi/WebFileSystem.cs [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+        {
+            if (!fileData.IsValidImage(fileName))
+            {
+                return false;
+            }
+
+            return await UploadToWeb(fileName, fileData);
+        }
+
+        public static async Task<bool> SavePicture(string pictureName, string pictureBase64)
```
Comment (ardalis) [inline src/PublicApi/WebFileSystem.cs:34]: Why is this static? You can't use it with DI or the IFileSystem interface.

File: src/PublicApi/WebFileSystem.cs [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+using System.Net;
+using System.Net.Http;
+using System.Text;
+using System.Threading.Tasks;
+using Microsoft.eShopWeb.PublicApi.CatalogItemEndpoints;
+using Newtonsoft.Json;
+
+namespace Microsoft.eShopWeb.PublicApi
+{
+    public class WebFileSystem: IFileSystem
```
Comment (ardalis) [inline src/PublicApi/WebFileSystem.cs:11]: Ideally the implementation belongs in the Infrastructure project and shouldn't be referenced directly from the front end project.

File: src/PublicApi/ImageValidators.cs (hunk 1 above)
Comment (ShadyNagy) [reply src/PublicApi/ImageValidators.cs]: Done

File: src/PublicApi/WebFileSystem.cs (hunk 3 above)
Comment (ShadyNagy) [reply src/PublicApi/WebFileSystem.cs:11]: PublicApi it is not in the frontend it is backend. so we need to move the WebFileSystem and IFileSystem to Infrastructure?

File: src/PublicApi/WebFileSystem.cs (hunk 2 above)
Comment (ShadyNagy) [reply src/PublicApi/WebFileSystem.cs]: Done

File: src/PublicApi/WebFileSystem.cs (hunk 3 above)
Comment (ShadyNagy) [reply src/PublicApi/WebFileSystem.cs:11]: IFileSystem to Infrastructure done

File: src/ApplicationCore/Entities/CatalogItem.cs [hunk 4]
```diff
@@ -54,5 +55,15 @@ public void UpdateType(int catalogTypeId)
...
         }
+
+        public void UpdatePictureUri(string pictureName)
+        {
+            if (string.IsNullOrEmpty(pictureName))
+            {
+                PictureUri = string.Empty;
+                return;
+            }
+            PictureUri = $"images\\products\\{pictureName}?{new DateTime().Ticks}";
```
Comment (ardalis) [inline src/ApplicationCore/Entities/CatalogItem.cs:66]: Is this just for cache busting? Is it necessary?

Comment (ShadyNagy) [reply src/ApplicationCore/Entities/CatalogItem.cs:66]: yes very necessary image will not refresh in the edit sfter change if this not there.

**Pull request merged**